include/measurement_kit/nettest.hpp

```diff
@@ -0,0 +1,1922 @@
+    static constexpr const char *key = "status.report_close";
+    
+    /// The "report_id" attribute of this event.
+    std::string report_id = {};
+};
+
+/// C++ representation of the "status.report_create" event.
+class StatusReportCreateEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.report_create";
+    
+    /// The "report_id" attribute of this event.
+    std::string report_id = {};
+};
+
+/// C++ representation of the "status.resolver_lookup" event.
+class StatusResolverLookupEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.resolver_lookup";
+    
+    /// The "ip_address" attribute of this event.
+    std::string ip_address = {};
+};
+
+/// C++ representation of the "status.started" event.
+class StatusStartedEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.started";
+    /* No attributes */
+};
+
+/// C++ representation of the "status.update.performance" event.
+class StatusUpdatePerformanceEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.update.performance";
+    
+    /// The "direction" attribute of this event.
+    std::string direction = {};
+    
+    /// The "elapsed" attribute of this event.
+    double elapsed = {};
+    
+    /// The "num_streams" attribute of this event.
+    int64_t num_streams = {};
+    
+    /// The "speed_kbps" attribute of this event.
+    double speed_kbps = {};
+};
+
+/// C++ representation of the "status.update.websites" event.
+class StatusUpdateWebsitesEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "status.update.websites";
+    
+    /// The "url" attribute of this event.
+    std::string url = {};
+    
+    /// The "status" attribute of this event.
+    std::string status = {};
+};
+
+/// C++ representation of the "task_terminated" event.
+class TaskTerminatedEvent {
+  public:
+    /// The unique identifier of this event.
+    static constexpr const char *key = "task_terminated";
+    /* No attributes */
+};
+
+} // namespace events
+
+/// Contains settings classes.
+namespace settings {
+
+/// Settings common to all nettests.
+class CommonSettings {
+  public:
+    /// String representation of the "err" log level.
+    static constexpr const char *log_level_err = "ERR";
+
+    /// String representation of the "warning" log level.
+    static constexpr const char *log_level_warning = "WARNING";
+
+    /// String representation of the "info" log level.
+    static constexpr const char *log_level_info = "INFO";
+
+    /// String representation of the "debug" log level.
+    static constexpr const char *log_level_debug = "DEBUG";
+
+    /// String representation of the "debug2" log level.
+    static constexpr const char *log_level_debug2 = "DEBUG2";
```

Comment:
The FFI API uses strings for verbosity. Since one typically needs to use such strings when setting the `Settings` I have decided to put them as static attributes of the `CommonSettings` class itself.